Platform/ADLINK/ComHpcAltPkg/Drivers/HardwareMonitorDxe/HwMonitorDxe.c

```diff
@@ -0,0 +1,488 @@
+/** @file
+
+  Copyright (c) 2020 - 2021, Ampere Computing LLC. All rights reserved.<BR>
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+
+#include <Guid/MdeModuleHii.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/DevicePathLib.h>
+#include <Library/HiiLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/PrintLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiLib.h>
+#include <Library/MmcLib.h>
+
+#include "HwMonitorHii.h"
+
+//
+// uni string and Vfr Binary data.
+//
+extern UINT8  HwMonitorVfrBin[];
+extern UINT8  HwMonitorDxeStrings[];
+
+EFI_HANDLE      mDriverHandle = NULL;
+EFI_HII_HANDLE  mHiiHandle    = NULL;
+
+#pragma pack(1)
+
+//
+// HII specific Vendor Device Path definition.
+//
+typedef struct {
+  VENDOR_DEVICE_PATH          VendorDevicePath;
+  EFI_DEVICE_PATH_PROTOCOL    End;
+} HII_VENDOR_DEVICE_PATH;
+
+#pragma pack()
+
+// HW_MONITOR_FORMSET_GUID
+EFI_GUID  gHardwareMonitorFormsetGuid = HW_MONITOR_FORMSET_GUID;

```

Comment:
gHardwareMonitorFormsetGuid should be defined in Platform/ADLINK/ComHpcAltPkg/ComHpcAlt.dec and extern it in Platform/ADLINK/ComHpcAltPkg/Drivers/HardwareMonitorDxe/HwMonitorHii.h

If it does not need a global definition, I think we can rename it to `mHardwareMonitorFormsetGuid`